contrib/ext_editor.lua

```diff
@@ -0,0 +1,371 @@
+local function OpenWith(images, choice)
+		
+	if #images == 1 then
+		for _, image in pairs(images) do 
+			
+			-- image to be edited
+			name = image.path..PS..image.filename 					
+			
+			-- launch the external editor
+			local run_cmd = df.sanitize_filename(program_paths[choice]).." "..df.sanitize_filename(name) 
+			dt.print("Launching "..program_names[choice].."...")	
+			dtsys.external_command(run_cmd)
+			
+			-- refresh the image view
+			-- note that only image:drop_cache() is not enough to refresh view in darkroom mode
+			-- therefore image must be deleted and reimported to force refresh
+			
+			local image_leader = image.group_leader
+			image:delete ()
+			local new_image = dt.database.import(name)
+			new_image:group_with(image_leader)
+
+			-- copy tags from image leader
+			for _,tag in pairs(dt.tags.get_tags(image_leader)) do
+				if not (string.sub(tag.name,1,9) == "darktable") then
+					dt.tags.attach(tag, new_image)
+					end
+				end
+			
+			-- remember the last editor used
+			dt.preferences.write("ext_editor","lastchoice", "integer", choice)
+			
+			-- refresh darkroom view
+			if dt.gui.current_view () == dt.gui.views.darkroom then
+				dt.gui.views.darkroom.display_image(new_image)
+				end
+			end
+	else
+		dt.print('please select one image')
+		end
+		
+	end
+
+
+-- callback for button "copy and edit"
+local function CopyAndOpenWith(images, choice)
+	
+	if #images == 1 then
+		for _,image in pairs(images) do 
+			
+			-- image to be copied and edited
+			local name = image.path..PS..image.filename
+			local new_name = name
+			
+			-- create unique filename
+			while df.check_if_file_exists(new_name) do
+				new_name = df.filename_increment(new_name)
+				end
+			
+			-- limit to 99 more exports of the original export
+			if string.match(df.get_basename(new_name), "_(d-)$") == "99" then
+					break
+				end
+				
+			-- physical copy
+			local result = df.file_copy(name, new_name)
+			
+			if result then
+				-- launch the external editor
+				local run_cmd = df.sanitize_filename(program_paths[choice]).." "..df.sanitize_filename(new_name)  	
+				dt.print("Launching "..program_names[choice].."...")	
+				dtsys.external_command(run_cmd)
+				
+				-- import in database and group
+				local new_image = dt.database.import(new_name)
+				new_image:group_with(image.group_leader)
+				
+				-- copy image tags
+				for _,tag in pairs(dt.tags.get_tags(image)) do
+					if not (string.sub(tag.name,1,9) == "darktable") then
+						dt.tags.attach(tag, new_image)
+						end
+					end
+				
+				-- remember the last editor used
+				dt.preferences.write("ext_editor","lastchoice", "integer", choice)
+				end	
+			end
+	else
+		dt.print('please select one image')
+		end
+		
+		
+	end
+
+
+-- callback function for shortcuts
+local function program_shortcut(event, shortcut)
+	local choice = tonumber(string.sub(shortcut, -1))
+	OpenWith(dt.gui.action_images, choice)
+	end
+
+
+-- export images and reimport in collection
```

Comment:
Is this to create an export from a raw so that the editor can be invoked on the export?